data/scripts/creaturescripts/player/login.lua

```diff
@@ -0,0 +1,162 @@
+	if (playerPos.x ~= oldPos.x or playerPos.y ~= oldPos.y or playerPos.z ~= oldPos.z) or player:getTarget() then
+		player:kv():remove("combat-protection")
+		return true
+	end
+
+	addEvent(onMovementRemoveProtection, 1000, playerId, oldPos, time - 1)
+end
+
+local playerLogin = CreatureEvent("PlayerLogin")
+
+function playerLogin.onLogin(player)
+	-- Welcome
+	local loginStr
+	if player:getLastLoginSaved() == 0 then
+		loginStr = "Please choose your outfit."
+		player:sendOutfitWindow()
+
+		db.query("UPDATE `players` SET `istutorial` = 0 WHERE `id` = " .. player:getGuid())
+	else
+		loginStr = string.format("Your last visit in %s: %s.", SERVER_NAME, os.date("%d %b %Y %X", player:getLastLoginSaved()))
+	end
+	player:sendTextMessage(MESSAGE_LOGIN, loginStr)
+
+	-- Promotion
+	local vocation = player:getVocation()
+	local promotion = vocation:getPromotion()
+	if player:isPremium() then
+		local hasPromotion = player:kv():get("promoted")
+
+		if not player:isPromoted() and hasPromotion then
+			player:setVocation(promotion)
+		end
+	elseif player:isPromoted() then
+		player:setVocation(vocation:getDemotion())
+	end
+
+	-- Boosted
+	player:sendTextMessage(MESSAGE_BOOSTED_CREATURE, "Today's boosted creature: " .. Game.getBoostedCreature() .. " \
+	Boosted creatures yield more experience points, carry more loot than usual and respawn at a faster rate.")
+	player:sendTextMessage(MESSAGE_BOOSTED_CREATURE, "Today's boosted boss: " .. Game.getBoostedBoss() .. " \
+	Boosted bosses contain more loot and count more kills for your Bosstiary.")
+
+	-- Rewards
+	local rewards = #player:getRewardList()
+	if rewards > 0 then
+		player:sendTextMessage(MESSAGE_LOGIN, string.format("You have %d %s in your reward chest.", rewards, rewards > 1 and "rewards" or "reward"))
+	end
+
+	-- Rate events:
+	if SCHEDULE_EXP_RATE ~= 100 then
+		sendBoostMessage(player, "Exp Rate", SCHEDULE_EXP_RATE > 100)
+	end
+
+	if SCHEDULE_SPAWN_RATE ~= 100 then
+		sendBoostMessage(player, "Spawn Rate", SCHEDULE_SPAWN_RATE > 100)
+	end
+
+	if SCHEDULE_LOOT_RATE ~= 100 then
+		sendBoostMessage(player, "Loot Rate", SCHEDULE_LOOT_RATE > 100)
+	end
+
+	if SCHEDULE_SKILL_RATE ~= 100 then
+		sendBoostMessage(player, "Skill Rate", SCHEDULE_SKILL_RATE > 100)
+	end
+
+	-- Send Recruiter Outfit
+	local playerName = getPlayerName(player)
+	local accountId = getAccountNumberByPlayerName(playerName)
+	local resultId = db.storeQuery("SELECT `recruiter` FROM `accounts` WHERE `id`=" .. accountId)
+	if resultId then
+		return true
+	end
+
+	local recruiterStatus = Result.getNumber(resultId, "recruiter")
+	local sex = player:getSex()
+	local outfitId = (sex == 1) and 746 or 745
+	for outfitAddOn = 0, 2 do
+		if recruiterStatus >= outfitAddOn * 3 + 1 then
+			local hasOutfit = player:hasOutfit(outfitId, outfitAddOn)
+			if not hasOutfit then
+				if outfitAddOn == 0 then
+					player:addOutfit(outfitId)
+				else
+					player:addOutfitAddon(outfitId, outfitAddOn)
+				end
+			end
+		end
+	end
```

Comment:
coloca a funcionalidade dentro do if. fica "mais seguro" e esse if está incorreto.
```suggestion
	local resultId = db.storeQuery("SELECT `recruiter` FROM `accounts` WHERE `id`=" .. accountId)
	if resultId then
	  local recruiterStatus = Result.getNumber(resultId, "recruiter")
	  local sex = player:getSex()
	  local outfitId = (sex == 1) and 746 or 745
	  for outfitAddOn = 0, 2 do
		if recruiterStatus >= outfitAddOn * 3 + 1 then
			local hasOutfit = player:hasOutfit(outfitId, outfitAddOn)
			if not hasOutfit then
				if outfitAddOn == 0 then
					player:addOutfit(outfitId)
				else
					player:addOutfitAddon(outfitId, outfitAddOn)
				end
			end
		end
	end
  end
```